Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -424,39 +424,96 @@ namespace ROS2::Utils
         VisitModelsForNestedModels(sdfRoot);
     }
 
-    AZStd::unordered_set<AZStd::string> GetMeshesFilenames(const sdf::Root& root, bool visual, bool colliders)
+    AssetFilenameReferences GetReferencedAssetFilenames(const sdf::Root& root)
     {
-        AZStd::unordered_set<AZStd::string> filenames;
-        auto GetMeshesFromModel = [&filenames, visual, colliders](const sdf::Model& model, const ModelStack&) -> VisitModelResponse
+        AssetFilenameReferences filenames;
+        auto GetMeshesFromModel = [&filenames](const sdf::Model& model, const ModelStack&) -> VisitModelResponse
```

Comment:
I guess we can rename this to "GetAssetsFromModel"?

Reply:
Ooops, missed that one, I'll fix.

Reply:
Fixed!